idioms/ctor.md

```diff
@@ -3,34 +3,74 @@
 ## Description
 
 Rust does not have constructors as a language construct. Instead, the
-convention is to use a static `new` method to create an object.
-
-## Example
-
-```rust,ignore
-// A Rust vector, see liballoc/vec.rs
-pub struct Vec<T> {
-    buf: RawVec<T>,
-    len: usize,
-}
-
-impl<T> Vec<T> {
-    // Constructs a new, empty `Vec<T>`.
-    // Note this is a static method - no self.
-    // This constructor doesn't take any arguments, but some might in order to
-    // properly initialise an object
-    pub fn new() -> Vec<T> {
-        // Create a new Vec with fields properly initialised.
-        Vec {
-            // Note that here we are calling RawVec's constructor.
-            buf: RawVec::new(),
-            len: 0,
-        }
+convention is to use an associated `new` function to create an object:
+
+```rust
+pub struct Second {
+    value: u64
+}
+
+impl Second {
+    pub fn new(value: u64) -> Self {
```

Comment:
Personally, I *really* dislike doc comments as they reduce readability of the actual code. That's why I removed them. Especially with an mdbook like this, where there can be context with the texts surrounding the examples, I'd prefer to have these notes in those texts rather than in the code examples. This keeps the code clean and tidy. 

If you want to highlight that an associated function doesn't use `self`, that's out of scope of the section. In this case, I would rather create a hyperlink on the word [associated](https://doc.rust-lang.org/reference/items/associated-items.html#associated-functions-and-methods) in the text above (like I did here) which points to the rust reference (or any other doc, like [the book](https://doc.rust-lang.org/stable/book/ch05-03-method-syntax.html#associated-functions)) explaining what associated functions are.

Reply:
> If you want to highlight that an associated function doesn't use self, that's out of scope of the section. In this case, I would rather create a hyperlink on the word associated in the text above (like I did here) which points to the rust reference (or any other doc, like the book) explaining what associated functions are.

Why not do both? :) A comment and a hyperlink in the text?